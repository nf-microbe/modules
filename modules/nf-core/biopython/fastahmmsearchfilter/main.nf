process BIOPYTHON_FASTAHMMSEARCHFILTER {
    tag "${meta.id}"
    label 'process_low'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/mulled-v2-80c23cbcd32e2891421c54d1899665046feb07ef:77a31e289d22068839533bf21f8c4248ad274b60-0' :
        'biocontainers/mulled-v2-80c23cbcd32e2891421c54d1899665046feb07ef:77a31e289d22068839533bf21f8c4248ad274b60-0' }"

    input:
    tuple val(meta) , path(fasta)
    tuple val(meta2), path(hmm_tbl)

    output:
    tuple val(meta), path("${prefix}.hmmfilter.fasta.gz")   , emit: fasta
    path "versions.yml"                                     , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    prefix = task.ext.prefix ?: "${meta.id}"
    """
    fastahmmsearchfilter.py \\
        --input_fasta ${fasta} \\
        --hmm_tbl ${hmm_tbl} \\
        --prefix ${prefix}.hmmfilter \\
        ${args}

    gzip ${prefix}.hmmfilter.fasta

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        python: \$( python --version | sed 's/Python //' )
        biopython: \$(python -c "import Bio; print(Bio.__version__)")
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    prefix = task.ext.prefix ?: "${meta.id}"
    """
    echo "" | gzip > ${prefix}.hmmfilter.fasta.gz

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        python: \$( python --version | sed 's/Python //' )
        biopython: \$(python -c "import Bio; print(Bio.__version__)")
    END_VERSIONS
    """
}
