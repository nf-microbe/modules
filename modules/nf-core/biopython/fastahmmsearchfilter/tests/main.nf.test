nextflow_process {

    name "Test Process: BIOPYTHON_FASTAHMMSEARCHFILTER"
    script "../main.nf"
    process "BIOPYTHON_FASTAHMMSEARCHFILTER"

    tag "modules_nfcore"
    tag "modules"
    tag "biopython"
    tag "biopython/fastahmmsearchfilter"

    test("fasta.gz + faa.gz") {
        when {
            process {
                """
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    file(params.modules_testdata_base_path + "genomics/prokaryotes/bacteroides_fragilis/illumina/fasta/test1.contigs.fa.gz", checkIfExists: true),
                ]
                input[1] = [
                    [ id: 'test' ],
                    file("${projectDir}/assets/test-datasets/genomics/provirus/illumina/hmmsearch/test.hmmsearch.tbl.gz", checkIfExists: true)
                ]
                """
            }
        }
        then {
            assertAll (
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

    test("stub") {
        when {
            process {
                """
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    file(params.modules_testdata_base_path + "genomics/prokaryotes/bacteroides_fragilis/illumina/fasta/test1.contigs.fa.gz", checkIfExists: true),
                ]
                input[1] = [
                    [ id: 'test' ],
                    file("${projectDir}/assets/test-datasets/genomics/provirus/illumina/hmmsearch/test.hmmsearch.tbl.gz", checkIfExists: true)
                ]
                """
            }
        }
        then {
            assertAll (
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }
}
