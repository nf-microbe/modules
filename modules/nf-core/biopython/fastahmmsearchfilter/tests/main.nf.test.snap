{
    "fasta.gz + faa.gz": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.hmmfilter.fasta.gz:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,c5a151d6b895998d0f58114062d9bc82"
                ],
                "fasta": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.hmmfilter.fasta.gz:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,c5a151d6b895998d0f58114062d9bc82"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.1",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-10-29T15:19:39.197107449"
    },
    "stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.hmmfilter.fasta.gz:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,c5a151d6b895998d0f58114062d9bc82"
                ],
                "fasta": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.hmmfilter.fasta.gz:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,c5a151d6b895998d0f58114062d9bc82"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.1",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-10-29T15:19:49.253953653"
    }
}